Assets/Scripts/Singleton.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Singleton<T> : MonoBehaviour where T : MonoBehaviour 
```

Comment:
since this is a mono behaviour you should handle the awake method for setting the instance = this

Reply:
Done.